src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -121,4 +121,14 @@ public void addMessage(Message message) {
   public void setMessages(List<Message> messages) {
     this.messages = messages;
   }
+
+  public List<Message> getMessagesByUserId(UUID author){
+    List<Message> ans = new ArrayList<Message>();
+    for(Message m : messages){
+      if(m.getAuthorId().equals(author)){
```

Comment:
Store in Map<UUID, List<Message>> (or a multimap, up to you) (by author UUID) so retrieval time is O(1) instead of O(N)